src/drivers/gnss/septentrio/septentrio.cpp

```diff
@@ -0,0 +1,1439 @@
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file septentrio.cpp
+ *
+ * Septentrio GNSS receiver driver
+ *
+ * @author Matej Franceskin <Matej.Franceskin@gmail.com>
+ * @author <a href="https://github.com/SeppeG">Seppe Geuens</a>
+ * @author Thomas Frans
+*/
+
+#include "septentrio.h"
+
+#include <cmath>
+#include <ctime>
+#include <string.h>
+#include <termios.h>
+#include <drivers/drv_hrt.h>
+#include <mathlib/mathlib.h>
+#include <matrix/math.hpp>
+#include <px4_platform_common/defines.h>
+#include <px4_platform_common/time.h>
+#include <uORB/topics/gps_inject_data.h>
+
+#include "util.h"
+
+using namespace time_literals;
+
+#ifndef GPS_READ_BUFFER_SIZE
+#define GPS_READ_BUFFER_SIZE 150 ///< Buffer size for `read()` call
+#endif
+
+// TODO: This functionality is not available on the Snapdragon yet.
+#ifdef __PX4_QURT
+#define NO_MKTIME
+#endif
+
+#define SBF_CONFIG_TIMEOUT      1000    ///< Timeout for waiting on ACK in ms
+#define SBF_PACKET_TIMEOUT      2       ///< If no data during this delay (in ms) assume that full update received
+#define DISABLE_MSG_INTERVAL    1000000 ///< Try to disable message with this interval (in us)
+#define MSG_SIZE                100     ///< Size of the message to be sent to the receiver
+#define TIMEOUT_5HZ             500     ///< Timeout time in mS, 1000 mS (1Hz) + 300 mS delta for error
+#define RATE_MEASUREMENT_PERIOD 5_s     ///< Rate at which bandwith measurements are taken
+#define RECEIVER_BAUD_RATE      115200  ///< The baudrate of the serial connection to the receiver
+// TODO: This number seems wrong. It's also not clear why an ULL is created and casted to UL (time_t).
+#define GPS_EPOCH_SECS ((time_t)1234567890ULL)
```

Comment:
I just copied this from `src/drivers/gps/devices/src/gps_helper.h` and changed the TODO a bit. If anyone has an idea what this does, I'd love to know 😄